File: src/com/l2kt/gameserver/network/clientpackets/RequestCursedWeaponList.java
package com.l2kt.gameserver.network.clientpackets;

import com.l2kt.gameserver.data.manager.CursedWeaponManager;
import com.l2kt.gameserver.model.actor.instance.Player;
import com.l2kt.gameserver.network.serverpackets.ExCursedWeaponList;

public class RequestCursedWeaponList extends L2GameClientPacket
{
	@Override
	protected void readImpl()
	{
	}
	
	@Override
	protected void runImpl()
	{
		final Player player = getClient().getActiveChar();
		if (player == null)
			return;
		
		player.sendPacket(new ExCursedWeaponList(CursedWeaponManager.getInstance().getCursedWeaponsIds()));
	}
}

File: src/com/l2kt/gameserver/network/clientpackets/RequestCursedWeaponList.kt
package com.l2kt.gameserver.network.clientpackets

import com.l2kt.gameserver.data.manager.CursedWeaponManager
import com.l2kt.gameserver.network.serverpackets.ExCursedWeaponList

class RequestCursedWeaponList : L2GameClientPacket() {
    override fun readImpl() {}

    override fun runImpl() {
        val player = client.activeChar ?: return
        player.sendPacket(ExCursedWeaponList(CursedWeaponManager.getInstance().cursedWeaponsIds))
    }
}